### PR TITLE
feat(grovedb,merk): no-prove query_aggregate_count_and_sum accumulator

### PR DESCRIPTION
**Pull request by QuantumExplorer**
## Summary

Adds the no-prove sibling of `Query::new_aggregate_count_and_sum_on_range`: `GroveDb::query_aggregate_count_and_sum` returns `(u64 count, i64 sum)` from a single merk-internal walk over a `ProvableCountProvableSumTree` (PCPS) terminator. Same call shape as `query_aggregate_sum` / `query_aggregate_count`, but yields both metrics at once.

## Why

Consumers that need both metrics on the no-proof path currently issue two parallel accumulator calls under a shared read transaction. Same cost class (two O(log n) merk walks), but two grovedb round-trips and two merk-internal traversals. Dash Platform's AVG no-prove dispatcher ([dashpay/platform#3690](https://github.com/dashpay/platform/pull/3690)) does this today — see `flat_aggregate_count_and_sum` at `packages/rs-drive/src/query/drive_document_count_and_sum_query/executors/range_no_proof.rs`. A combined no-prove primitive collapses those into one call, mirroring what the proof-side combined primitive already does for the prove path.

## What changed

- **`merk/.../aggregate_count_and_sum/walk.rs`** (new) — `walk_count_and_sum` classification walker accumulating `(u128, i128)` end-to-end so adversarial intermediate states cannot wrap mid-walk.
- **`merk/.../aggregate_count_and_sum/prove.rs`** — `RefWalker::count_and_sum_aggregate_on_range` narrowing to `(u64, i64)` with the same overflow check `sum_aggregate_on_range` uses.
- **`merk/src/merk/get.rs`** — `Merk::count_and_sum_aggregate_on_range` with PCPS-only tree-type gate (same shape the proof-side primitive rejects single-axis hosts with) and `(0, 0)` on empty merk.
- **`grovedb/src/operations/get/query.rs`** — `GroveDb::query_aggregate_count_and_sum` mirroring `query_aggregate_sum`'s body. Up-front leaf-shape validation via the pre-existing `validate_leaf_aggregate_count_and_sum_on_range`, opens leaf merk, delegates, wraps merk error with path context.
- **`grovedb-version`** — new `query_aggregate_count_and_sum_on_range` slot (`0` in v1/v2/v3) — V0-routed like the count/sum siblings.

## Tests

- **10 merk-level no-prove tests** covering each `Range*` variant, full-range happy path, empty merk, disjoint range, mixed-sign sums, and PCPS-only enforcement across every other tree type. Each cross-checks the no-prove walker against the prove-side primitive on the same merk so the two paths cannot silently diverge.
- **16 grovedb-level no-prove tests** covering the same range-shape sweep end-to-end, plus carrier-shape rejection (the leaf-only validator must fire even when the dispatcher would accept the same query), empty-path rejection, `NormalTree`-at-merk rejection, single-axis `ProvableCountSumTree` rejection, and `GROVE_V2` version-gate routing. Each fixture call asserts `(count, sum)` against what the prove-side verifier extracts for the same path query, pinning prove/no-prove equivalence the way the existing sum/count tests do.

## Out of scope

- Carrier (`query_aggregate_count_and_sum_per_key`) variant — not needed for Dash Platform's AVG no-prove dispatcher (the carrier-AVG case is prove-only) and explicitly deferred to a separate PR. The entry-point docstring notes this.
- No changes to `Query::new_aggregate_count_and_sum_on_range`, the proof-side primitive, the verifier, or the carrier shape.

## Test plan

- [x] `cargo test -p grovedb-merk --features minimal` — 667 passed (10 new)
- [x] `cargo test -p grovedb --features minimal` — 1849 passed (16 new)
- [x] `cargo clippy -p grovedb-merk --features minimal -- -D warnings` — clean
- [x] `cargo clippy -p grovedb --features minimal -- -D warnings` — clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Introduced a new query operation that retrieves both count and sum statistics in a single execution across a specified data range, eliminating the need for separate count and sum queries.

* **Tests**
  * Added comprehensive test coverage for the new operation, including inclusive/exclusive range tests, edge cases with empty trees and mixed values, version compatibility validation, and error handling scenarios.

[Review Change Stack](https://app.coderabbit.ai/change-stack/dashpay/grovedb/pull/676?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: grovedb/src/tests/aggregate_count_and_sum_query_tests.rs [hunk 1]
```diff
@@ -0,0 +1,521 @@
...
+        );
+        let direct = db
+            .grove_db
+            .query_aggregate_count_and_sum(&path_query, None, v)
+            .unwrap()
+            .expect("query under GROVE_V2 should succeed (slot is 0)");
+        // Spot-check the count axis on the full fixture; sum details
+        // are covered by the latest-version tests above.
+        assert_eq!(direct.0, 15);
+    }
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply grovedb/src/tests/aggregate_count_and_sum_query_tests.rs]: This is Claude. Renamed to `no_proof_combined_v0_envelope_accepted_under_v2` (and adjusted the docstring) so the name matches the success assertion — pushed in 77a1fd9.

**Review by QuantumExplorer — COMMENTED** [review]
Reviewed
